augur/dates.py

```diff
@@ -43,7 +43,7 @@ class InvalidDateFormat(ValueError):
 # e.g. 2018
 # and other positive ints
 # e.g. 1, 123, 12345
-RE_YEAR_ONLY = re.compile(r'^[1-9][0-9X]*$')
+RE_YEAR_ONLY = re.compile(r'^0*[1-9][0-9X]*$')
```

Comment:
Just a note that I was wondering about if could simplify down to `^[0-9X]+$`, but I think I realized why not: neither `0` nor `X` alone should be allowed, and the `[1-9]` in the middle ensures this. (However, it also means that `0XXX` and `XX00` don't match.)

Reply:
> However, it also means that `0XXX` and `XX00` don't match.

Hmm I hadn't thought of this.

- `0XXX`: this is just the same as `XXX` which I don't think should be supported.
- `XX00`: this should be supported if `XX11` is also supported (it is currently matched). However, if in [#989](https://github.com/nextstrain/augur/issues/989) the decision is to stop supporting partial ambiguity in years, then neither `XX00` nor `XX11` should be supported.

Reply:
6a18b35c15b7892deef4a7baff3b445dab8a4a81 removes partial ambiguity so this isn't an issue anymore.